joint_trajectory_controller/src/joint_trajectory_controller.cpp

```diff
@@ -608,29 +608,33 @@ CallbackReturn JointTrajectoryController::on_configure(const rclcpp_lifecycle::S
       return CallbackReturn::FAILURE;
     }
   }
-  else if (has_acceleration_state_interface_)
-  {
-    RCLCPP_ERROR(
-      logger,
-      "'acceleration' state interface cannot be used if 'position' and 'velocity' "
-      "interfaces are not present.");
-    return CallbackReturn::FAILURE;
-  }
-  else if (has_velocity_command_interface_ && command_interface_types_.size() == 1)
-  {
-    RCLCPP_ERROR(
-      logger,
-      "'velocity' command interface can only be used alone if 'velocity' and "
-      "'position' state interfaces are present");
-    return CallbackReturn::FAILURE;
-  }
-  else if (has_effort_command_interface_)
+  else
   {
-    RCLCPP_ERROR(
-      logger,
-      "'effort' command interface can only be used alone if 'velocity' and "
-      "'position' state interfaces are present");
-    return CallbackReturn::FAILURE;
+    if (has_acceleration_state_interface_)
+    {
+      RCLCPP_ERROR(
+        logger,
+        "'acceleration' state interface cannot be used if 'position' and 'velocity' "
+        "interfaces are not present.");
+      return CallbackReturn::FAILURE;
+    }
+    if (has_velocity_command_interface_ && command_interface_types_.size() == 1)
+    {
+      RCLCPP_ERROR(
+        logger,
+        "'velocity' command interface can only be used alone if 'velocity' and "
+        "'position' state interfaces are present");
+      return CallbackReturn::FAILURE;
+    }
+    // effort is always used alone so no need for size check
+    if (has_effort_command_interface_)
```

Comment:
This should have failed on tests… because always when you are having “effort” command interface this will fail. We should at least check if there is only one interface presents, as done for the velocity. But I would be more happy if we make this checks more explicit.

They are correct, but unclear/confusing.